internal/util.go

```diff
@@ -12,13 +12,17 @@ import (
 // for resources that can contain an arbitrary or dynamic number of fields.
 func RemainingKeys(s interface{}, m map[string]interface{}) (extras map[string]interface{}) {
 	extras = make(map[string]interface{})
+	for k, v := range m {
+		extras[k] = v
+	}
+
 	valueOf := reflect.ValueOf(s)
 	typeOf := reflect.TypeOf(s)
 	for i := 0; i < valueOf.NumField(); i++ {
 		field := typeOf.Field(i)
 		tagValue := field.Tag.Get("json")
-		if _, ok := m[tagValue]; !ok {
-			extras[tagValue] = m[tagValue]
+		if _, ok := extras[tagValue]; ok {
+			delete(extras, tagValue)
```

Comment:
> However, the function is supposed to collect fields that have no json tags

I think it should be different than that; namely, I think we want a function that will return keys that are in the response JSON `map` but are not fields in the defined `struct` response object. `UnmarshalJSON` can work just fine without struct tags.
For example, in your unit test, `City` is a known field, so I wouldn't expect it to get returned from this function.